### PR TITLE
CVE-2016-7401

### PR DESCRIPTION
**Pull request by zxw4320**

### CODE REVIEW

File: cves/CVE-2016-7401.yml [hunk 1]
```diff
@@ -66,30 +68,24 @@ fixes_vcc_instructions: |
...
-  note: This VCC was discovered automatically via archeogit.
-- commit: 75536fef1ff56894e35dad221a4215ebe3917673
-  note: This VCC was discovered automatically via archeogit.
+  note: Moved cookie class.
+- commit: 088f7170779b98547d9862603387afee77b964a8
+  note: Added cookie class.
+- commit: 2659429df46d0d79ad7a93d50d0ccdab45b55b53
+  note: |
+    The original VCC was part of modifying how cookies were passed
+    as a bug prevented them from being passed through load.
```
Comment (rxv3774) [inline cves/CVE-2016-7401.yml]: Changing this sentence to something like "The original VCC contained changes that modified the way in which cookies were being passed, as a bug was preventing them from being passed through load" would make it a bit easier to understand.

File: cves/CVE-2016-7401.yml [hunk 2]
```diff
@@ -111,10 +107,12 @@ unit_tested:
 
     For the fix_answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  code:
-  code_answer:
-  fix:
-  fix_answer:
+  code: true
+  code_answer: |
+    There were unit tests for cookies originally, but they failed to catch this
```
Comment (rxv3774) [inline cves/CVE-2016-7401.yml:115]: Describing what the original unit tests were testing for might help explain to the reader why this bug was not caught.

File: cves/CVE-2016-7401.yml [hunk 3]
```diff
@@ -66,30 +68,24 @@ fixes_vcc_instructions: |
...
-  note: This VCC was discovered automatically via archeogit.
-- commit: d63ee2097f4d0f0094e8e472de2236558e98a348
-  note: This VCC was discovered automatically via archeogit.
-- commit: 0ed7d155635da9f79d4dd67e4889087d3673c6da
-  note: This VCC was discovered automatically via archeogit.
-- commit: 75536fef1ff56894e35dad221a4215ebe3917673
-  note: This VCC was discovered automatically via archeogit.
+  note: Moved cookie class.
+- commit: 088f7170779b98547d9862603387afee77b964a8
+  note: Added cookie class.
```
Comment (rxv3774) [inline cves/CVE-2016-7401.yml]: Switching the order of the VCC commits to the following would make it easier to read and understand:
Added cookie class
Moved cookie class
Moved cookie code out of __init__.py and into cookie class
This is not a big issue, but just a small formatting detail that would make it a little cleaner.

File: cves/CVE-2016-7401.yml [hunk 4]
```diff
@@ -66,30 +68,24 @@ fixes_vcc_instructions: |
...
+- commit: 2659429df46d0d79ad7a93d50d0ccdab45b55b53
+  note: |
+    The original VCC was part of modifying how cookies were passed
+    as a bug prevented them from being passed through load.
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (rxv3774) [inline cves/CVE-2016-7401.yml:92]: upvotes: 7

Comment (zxw4320) [reply cves/CVE-2016-7401.yml:92]: updated

File: cves/CVE-2016-7401.yml (hunk 3 above)
Comment (zxw4320) [reply cves/CVE-2016-7401.yml]: I am keeping the order so that it is chronological, but I have updated the notes for clarity.

File: cves/CVE-2016-7401.yml (hunk 2 above)
Comment (zxw4320) [reply cves/CVE-2016-7401.yml:115]: Updated to state that some sanitization checks were missing along with testing for this edge case.

File: cves/CVE-2016-7401.yml (hunk 1 above)
Comment (zxw4320) [reply cves/CVE-2016-7401.yml]: Added clarification.
